### DISCUSSION TITLE
memory usage problem when collection  with IVF_FLAT index loaded

### DISCUSSION THREAD

**Discussion by wzyaq001** [milvus-io/milvus#35503] [Q&A and General discussion] [answered]
i have about 7 million vector data with dim 768, using IVF_FLAT index and n_list is 1024. When load the collection, i found memory useage first increases to 40GB, then descrease to 20GB. Given the calculation, 7000000 count * 8 byte * 768 dim = 40GB is raw data usage, 1024 * 8 * 768 = 0.01GB and 7000000 * 4 = 0.026GB is IVF index usage, so how is 20GB calculated?

**Comment by yhmo** [comment]
Each dim is a float32 value stored in milvus. So, each dim occupies 4 bytes space.
7M, 768dim vectors, the size is 7M * 768 * 4Bytes = 20GB.

Similar question:
https://github.com/milvus-io/milvus/discussions/32224

**Reply by wzyaq001** [reply]
oh,yes! i used FLOAT_VECTOR that stores 32-bit value. but i'm now confused why memory used 40 GB at first, my collection only has id and vector fields.

**Reply by yhmo** [reply, marked as answer]
Your milvus is standalone, all the components are running in one process, including the query nodes, data nodes and index nodes. There might be unfinished index tasks or compaction tasks at that time. After the tasks are finished, the memory usage decreases to 20GB.
